### PR TITLE
Fixes and changes to spin up a brand new atmos stack

### PR DESCRIPTION
**Pull request by iliazlobin**
## what
- remote_state invocation through the module, removing old code where remote state is pulled through data sources
- updated modules' versions in several places
- changes to providers.tf for bootstraping components with a direct assume role variable (we haven't figured out how to make it work through iam_roles module while we don't have any roles provisioned)
- missing and required variables that have references in module's code
- we'd be rather using roles instead of profiles when setting up AWS providers
- updated policies for ALB controller according to updated AWS docs
- created custom EKS IAM roles

### CODE REVIEW

File: modules/sso/providers.tf [hunk 1]
```diff
@@ -1,25 +1,12 @@
...
 
-  dynamic "assume_role" {
-    for_each = module.iam_roles.org_role_arn != null ? [true] : []
-    content {
-      # `terraform import` will not use data from a data source,
-      # so on import we have to explicitly specify the role
-      role_arn = coalesce(var.import_role_arn, module.iam_roles.org_role_arn)
-    }
+  assume_role {
+    role_arn = var.aws_assume_role_arn
```
Comment (Nuru) [inline modules/sso/providers.tf:5]: All the changes to this file should be reverted

File: modules/vpc-flow-logs-bucket/providers.tf [hunk 2]
```diff
@@ -2,17 +2,11 @@ provider "aws" {
   region = var.region
 
   assume_role {
-    role_arn = coalesce(var.import_role_arn, module.iam_roles.terraform_role_arn)
+    role_arn = var.aws_assume_role_arn
```
Comment (Nuru) [inline modules/vpc-flow-logs-bucket/providers.tf:5]: All the changes to this file should be reverted

File: modules/vpc/providers.tf [hunk 3]
```diff
@@ -2,12 +2,14 @@ provider "aws" {
...
   assume_role {
+    # `terraform import` will not use data from a data source,
+    # so on import we have to explicitly specify the role
     role_arn = coalesce(var.import_role_arn, module.iam_roles.terraform_role_arn)
   }
 }
 
 module "iam_roles" {
-  source  = "../account-map/modules/iam-roles"
+  source = "../account-map/modules/iam-roles"
```
Comment (Nuru) [inline modules/vpc/providers.tf:12]: Revert

**Review by Nuru — CHANGES_REQUESTED** [review]

**Comment by Nuru** [comment]
I rejected the changes to the providers for the bootstrap components because they create or modify IAM roles and/or authentication so you cannot use the roles they create to create them (which is why they are called "bootstrap" components).

File: modules/eks/versions.tf [hunk 4]
```diff
@@ -12,7 +12,7 @@ terraform {
     }
     kubernetes = {
       source  = "hashicorp/kubernetes"
-      version = "~> 1.13"
+      version = "~> 2.2.0"
```
[the remark this answers is not in the document]
Comment (iliazlobin) [reply modules/eks/versions.tf:15]: We were able to deploy the eks component with version 2.2.0 quite successfuly. Could it be that the problem was related to some of the previous versions of the k8s provider (2.1.x) that's been fixed in 2.2.0?

**Comment by sgtoj** [comment]
Maybe I am mistaken, but I don think you meant to commit...
```
modules/tfstate-backend/gbl-root-tfstate-backend.planfile
```

**Comment by nitrocode** [comment]
I think this was included on accident as well `gbl-root-tfstate-backend.tfplan`.

File: modules/efs/main.tf [hunk 5]
```diff
@@ -6,7 +6,7 @@ locals {
 }
 
 module "efs" {
-  source = "git::https://github.com/cloudposse/terraform-aws-efs.git?ref=tags/0.21.0"
+  source = "git::https://github.com/cloudposse/terraform-aws-efs.git?ref=tags/0.30.1"
```
Comment (nitrocode) [inline modules/efs/main.tf:9]: nit: We should use the more popular `source`

```suggestion
  source  = "cloudposse/efs/aws"
  version = "0.30.1"
```

File: modules/efs/remote-state.tf [hunk 6]
```diff
@@ -9,6 +9,7 @@ module "vpc" {
 }
 
 module "eks" {
+  count = var.use_eks_security_group ? 1 : 0
```
Comment (nitrocode) [inline modules/efs/remote-state.tf:12]:
```suggestion
  count   = var.use_eks_security_group ? 1 : 0
```

**Comment by Benbentwo** [comment]
Closed this as it has become stale, most of the components referenced by this branch and PR have been superseded or updated.

**Pull request closed without merging**